src/sentry/scim/endpoints/teams.py

```diff
@@ -0,0 +1,279 @@
+        ).order_by("slug")
+
+        if filter_val:
+            queryset = queryset.filter(slug=slugify(filter_val))
+
+        def data_fn(offset, limit):
+            return list(queryset[offset : offset + limit])
+
+        def on_results(results):
+            results = serialize(
+                results, None, TeamSCIMSerializer(), exclude_members=exclude_members
+            )
+            return {
+                "schemas": [SCIM_API_LIST],
+                "totalResults": queryset.count(),  # TODO: audit perf
+                "startIndex": int(request.GET.get("startIndex", 1)),  # must be integer
+                "itemsPerPage": len(results),  # what's max?
+                "Resources": results,
+            }
+
+        return self.paginate(
+            request=request,
+            on_results=on_results,
+            paginator=GenericOffsetPaginator(data_fn=data_fn),
+            default_per_page=int(request.GET.get("count", 100)),
+            queryset=queryset,
+            cursor_cls=SCIMCursor,
+        )
+
+    def post(self, request, organization):
+        """
+        Creates a team based on a SCIM Group POST request.
+        Basically the same operations as the regular team POST but with
+        SCIM serializers
+        """
+        serializer = SCIMTeamNameSerializer(data={"name": request.data["displayName"]})
+
+        if serializer.is_valid():
+            result = serializer.validated_data
+
+            try:
+                with transaction.atomic():
+                    team = Team.objects.create(
+                        name=result.get("name"),
+                        organization=organization,
+                    )
+                    self.create_audit_entry(
+                        request=request,
+                        organization=organization,
+                        target_object=team.id,
+                        event=AuditLogEntryEvent.TEAM_ADD,
+                        data=team.get_audit_log_data(),
+                    )
+            except IntegrityError:
+                return Response(
+                    {
+                        "non_field_errors": [CONFLICTING_SLUG_ERROR],
+                        "detail": CONFLICTING_SLUG_ERROR,
+                    },
+                    status=409,
+                )
+            else:
+                team_created.send_robust(
+                    organization=organization, user=request.user, team=team, sender=self.__class__
+                )
+
+            context = serialize(team, serializer=TeamSCIMSerializer())
+            return Response(context, status=201)
+        else:
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class OrganizationSCIMTeamDetails(SCIMEndpoint, OrganizationEndpoint):
+    permission_classes = (OrganizationSCIMTeamPermission,)
+
+    def convert_args(self, request, organization_slug, team_id, *args, **kwargs):
```

Comment:
can't inherit from the teams route that already exists as this route needs to be based on team_id and not team_slug, so built method to get team from team_id here.